src/contents/util/courseLogic.ts

```diff
@@ -237,31 +236,37 @@ const splitByNumbers = (input: string): string[] => {
 
 export const createSyllabusButton = async () => {
   if (!location.href.startsWith("https://scombz.shibaura-it.ac.jp/lms/course?")) return;
-  const currentData = (await chrome.storage.local.get(defaultSaves)) as Saves;
-  const fac = currentData.settings.faculty;
+
   const insertArea =
     document.querySelector(".contents-question-template-area") || document.querySelector(".course-title-txt");
-  if (!fac) {
-    insertArea.insertAdjacentHTML(
-      "afterend",
-      `<span style="color:red;padding: 12px 30px 10px 34px">
-        ${chrome.i18n.getMessage("openExtensionOptionDescription1")}
-        <a href="javascript:void(0);" id="link_to_extention_syll">${chrome.i18n.getMessage("extensionOption")}</a>
-        ${chrome.i18n.getMessage("openExtensionOptionDescription2")}
-        </span>
-    `,
-    );
-    document.getElementById("link_to_extention_syll")?.addEventListener("click", () => {
-      chrome.runtime.sendMessage({ action: "openOption" } as RuntimeMessage);
-    });
-    return;
+
+  const year = location.href.split("idnumber=")[1].slice(0, 4);
+  const facNumber = location.href.split("idnumber=")[1].slice(4, 6);
+
+  let fac = "";
+  if (facNumber === "01") {
+    fac = "ko1";
+  } else if (facNumber === "02") {
+    fac = "arc";
+  } else if (facNumber === "03") {
+    fac = "sys";
+  } else if (facNumber === "04") {
+    fac = "dsn";
+  } else if (facNumber === "05") {
+    fac = "din";
+  } else {
+    // 博士の番号が違う場合でもシラバス検索は変わらない？
+    console.log("createSyllabusButton: 学部特定失敗");
+    console.log();
+    fac = "din";
   }
```

Comment:
冗長に感じました。
こんな感じでどうでしょう？(動作未検証)
JSは配列の範囲外読み出しはundefinedが帰ってくるので、修正前と等価なコードのはずです。(elseの場合はundefinedを入れるようにだけ変更しました。)

```suggestion
  const year: string = location.search.split("idnumber=")[1].slice(0, 4);
  const facNumber: number = parseInt(location.search.split("idnumber=")[1].slice(4, 6), 10);
  const facArray = ["ko1", "arc", "sys", "dsn", "din"];
  const fac = Number.isNaN(facNumber) ? undefined : facArray[facNumber - 1];
```